core/server/master/src/main/java/alluxio/master/block/DefaultBlockMaster.java

```diff
@@ -672,6 +688,9 @@ public long getWorkerId(WorkerNetAddress workerNetAddress) {
         lostWorker.updateLastUpdatedTimeMs();
         mWorkers.add(lostWorker);
         mLostWorkers.remove(lostWorker);
+        for (Consumer<Long> function : mAddConfCallbacks) {
+          function.accept(lostWorker.getId());
```

Comment:
how will we handle the case where a worker is restarted with new configuration?

Reply:
If a worker gets its old id and restarts with new configuration, it will register with new configuration. When this worker register, we will overwrite its previous configuration in the confMap and remove its old configuration in lostConfMap if exists.

If the worker gets a new id and register with new configuration, the previous old id worker will be detected as lost worker and move to LostConfMap. new worker id alone with the worker configuration will be add to ConfMap.